### PR TITLE
DIV-4814 Added permissions for LA

### PR DESCRIPTION
**Pull request by ravinderpal**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DIV-4814

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: definitions/divorce/json/CaseType.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   {
     "LiveFrom": "01/01/2017",
     "ID": "DIVORCE",
-    "Name": "Divorce case - v113.19",
+    "Name": "Divorce case - v113.20",
```
Comment (hosainnet) [inline definitions/divorce/json/CaseType.json]: FYI you may have to up this depending on other PRs

File: definitions/divorce/json/AuthorisationCaseState.json [hunk 2]
```diff
@@ -831,5 +831,194 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseStateID": "solicitorAwaitingPaymentConfirmation",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseStateID": "SOTAgreementPayAndSubmitRequired",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseState.json]: As far as I can tell, `SOTAgreementPayAndSubmitRequired` didn't have the beta role

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 3]
```diff
@@ -3169,5 +3169,1153 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "reissueFromAosStarted",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "reissueFromIssuedRDC",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Don't think beta has this

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 4]
```diff
@@ -3169,5 +3169,1153 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "co-RespAwaitingLAReferral",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "testAwaitingDecreeAbsolute",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Don't think beta has this

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 5]
```diff
@@ -3169,5 +3169,1153 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "testAwaitingDecreeNisi",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "linkRespondent",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Don't think beta has this

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 6]
```diff
@@ -3169,5 +3169,1153 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "testAosAwaiting",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "testAwaitingDecreeNisi",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Don't think beta has this

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 7]
```diff
@@ -3169,5 +3169,1153 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "AddNote",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "testAosStarted",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Don't think beta has this

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 8]
```diff
@@ -3169,5 +3169,1153 @@
...
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "solicitorStatementOfTruthPaySubmit",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "23/05/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "solicitorUpdate",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Don't think beta has this

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 9]
```diff
@@ -9301,5 +9301,1979 @@
...
+    "LiveFrom": "25/04/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "DecreeNisiGrantedDate",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "DNApprovalDate",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseField.json]: It appears this is duplicated in the file

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 10]
```diff
@@ -9301,5 +9301,1979 @@
...
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "CostsClaimGranted",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseField.json]: It appears this is duplicated in the file

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 11]
```diff
@@ -9301,5 +9301,1979 @@
...
+    "LiveFrom": "01/01/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "BulkListingCaseId",
+    "UserRole": "caseworker-divorce-courtadmin-la",
+    "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "01/01/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseFieldID": "labelOutcomeofDN",
```
Comment (hosainnet) [inline definitions/divorce/json/AuthorisationCaseField.json]: It appears this is duplicated in the file

File: definitions/divorce/json/AuthorisationCaseField.json [hunk 12]
```diff
@@ -9301,5 +9301,1979 @@
     "CaseFieldID": "labelOutcomeofDN",
     "UserRole": "caseworker-divorce-superuser",
     "CRUD": "R"
+  },
```
Comment (qzhou-hmcts) [inline definitions/divorce/json/AuthorisationCaseField.json]: Why all the fields has CRU?

**Comment by hosainnet** [comment]
Sidenote, as part of #13 I made sure that the role exists in _both_ IDAM and CCD, in all env (demo, aat and even prod). Also I created a user on AAT, which has the new superuser role, and added it to `UserProfile.json` so that we can easily test the different roles we have

**Comment by qzhou-hmcts** [comment]
@vianvi how long do you think will take to prepare a test for this?

**Comment by hosainnet** [comment]
@qzhou-hmcts speaking of that, I think at some point we should move tests should move here to make sure they run with every PR

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 12 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseField.json]: it reflects the Field permission of `caseworker-divorce-courtadmin_beta`(as for AC 1)

File: definitions/divorce/json/AuthorisationCaseState.json (hunk 2 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseState.json]: yes, but the Case State (AC 2) doesn't reflect beta role, so I'm adding to all the distinct field that I've found @hosainnet

File: definitions/divorce/json/CaseType.json (hunk 1 above)
Comment (ravinderpal) [reply definitions/divorce/json/CaseType.json]: That's fine. I'll ask @vianvi to let me change this before merging

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 9 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseField.json]: removed

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 10 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseField.json]: removed a copy of it

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 11 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseField.json]: removed a copy of it

File: definitions/divorce/json/AuthorisationCaseEvent.json (hunk 3 above)
Comment (ravinderpal) [reply definitions/divorce/json/AuthorisationCaseEvent.json]: Case State (AC 2) refers to ALL states, not only the beta ones

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 12 above)
Comment (qzhou-hmcts) [reply definitions/divorce/json/AuthorisationCaseField.json]: I thought  we copy  `caseworker-divorce-courtadmin_beta` but with read permissions excluding these fields that the LA really need to update

File: definitions/divorce/json/AuthorisationCaseField.json (hunk 9 above)
Comment (tchow8) [reply definitions/divorce/json/AuthorisationCaseField.json]: Did you reintroduce it by accident?
https://github.com/hmcts/div-ccd-definitions/blob/017f89f11ec841169d4edd46ec3f0b4ce0c7158e/definitions/divorce/json/AuthorisationCaseField.json#L7229
Or am I looking at an old commit.

**Pull request merged**